tests/site/test_settings_helpers.py

```diff
@@ -37,3 +43,112 @@ def test_dont_strip_unknown_ports(self):
+                "path": "fghfh",
+                "urlpatterns_module": "django",  # must be a dotted path to a module in python path!
+            },
+        }
+        setting = AppSetting(**raw_setting)
+        result = get_urlpatterns_from_setting([setting])
+        self.assertEqual(len(result), 1)
+        self.assertTrue(isinstance(result[0], URLResolver))
+        self.assertFalse(result[0].namespace)
+
+    def test_urls_with_namespace_return_list_of_paths_with_namespace(self):
+        # django.urls.include has one obligatory positional argument that is
+        # either an urlpatterns-list or a dotted path to a module that contains
+        # an urlpatterns list that is labeled "urlpatterns". It also has
+        # a keyword argument "namespace". This tests including urlpatterns that
+        # DO have namespace set.
+        raw_setting = {
+            "app_name": "foo",
+            "urls": {
+                "path": "fghfh",
+                "urlpatterns_module": "django",  # must be a dotted path to a module in python path!
+                "namespace": "blbl",
+            },
+        }
+        setting = AppSetting(**raw_setting)
+        result = get_urlpatterns_from_setting([setting])
+        self.assertEqual(len(result), 1)
+        self.assertTrue(isinstance(result[0], URLResolver))
+        self.assertEqual(result[0].namespace, "blbl")
+
+
+class UpdateContextProcessorsListTests(unittest.TestCase):
+    def test_when_context_processor_setting_is_unset_it_should_do_nothing(self):
+        raw_setting = {
+            "app_name": "foo",
+            "urls": {
+                "path": "fghfh",
+                "urlpatterns_module": "django",  # must be a dotted path to a module in python path!
+                "namespace": "blbl",
+            },
+            # NO "context_processors"-key!
+        }
+        TEMPLATES = []
+        result = update_context_processors_list(TEMPLATES, None)
+        self.assertEqual(result, TEMPLATES)
+
+    def test_when_template_setting_is_falsey_it_should_do_nothing(self):
+        TEMPLATES = []
+        raw_setting = {
+            "app_name": "foo",
+            "context_processors": ["omega"],
+        }
+        app_setting = AppSetting(**raw_setting)
+        result = update_context_processors_list(TEMPLATES, [app_setting])
+        self.assertEqual(result, TEMPLATES)
+
+    def test_when_it_is_not_a_DjangoTemplates_section_it_should_do_nothing(self):
+        raw_setting = {
+            "app_name": "foo",
+            "context_processors": ["omega"],
+        }
+        app_setting = AppSetting(**raw_setting)
+        TEMPLATES = [
+            {"BACKEND": "django.template.backends.jinja2.Jinja2", "OPTIONS": {"context_processors": ["alpha"]}}
+        ]
+        result = update_context_processors_list(TEMPLATES, [app_setting])
+        self.assertEqual(result, TEMPLATES)
+
+    def test_append_context_processors_in_setting_to_DjangoTemplates_context_processors_list(self):
```

Comment:
```suggestion
    def test_when_app_settings_contain_context_processors_it_should_append_them_to_DjangoTemplates_context_processors_list(self):
```

Yes, it's long, but I don't care - clarity matters :)